packages/astro/src/vite-plugin-astro/index.ts

```diff
@@ -37,23 +46,37 @@ export default function astro({ config, logging }: AstroPluginOptions): vite.Plu
 			let { filename, query } = parseAstroRequest(id);
 			if (query.astro) {
 				if (query.type === 'style') {
-					if (filename.startsWith('/@fs')) {
-						filename = filename.slice('/@fs'.length);
-					} else if (filename.startsWith('/') && !ancestor(filename, config.projectRoot.pathname)) {
-						filename = new URL('.' + filename, config.projectRoot).pathname;
-					}
-					const transformResult = await cachedCompilation(config, filename, null, viteTransform, opts);
-
 					if (typeof query.index === 'undefined') {
 						throw new Error(`Requests for Astro CSS must include an index.`);
 					}
 
+					const transformResult = await cachedCompilation(config,
+						normalizeFilename(filename), null, viteTransform, opts);
 					const csses = transformResult.css;
 					const code = csses[query.index];
 
 					return {
 						code,
 					};
+				} else if(query.type === 'script') {
+					if(typeof query.index === 'undefined') {
+						throw new Error(`Requests for hoisted scripts must include an index`);
+					}
+
+					const transformResult = await cachedCompilation(config,
+						normalizeFilename(filename), null, viteTransform, opts);
+					const scripts = transformResult.scripts;
+					const r = scripts[query.index];
```

Comment:
What is r? Comment or rename would be helpful!

Reply:
Renamed, good suggestion!